File: src/java/Webapp/usr/fne/golfer/Applet/SigmaHoleInfo.java
//----------------------------------------------------------------------------
//
//       Copyright (C) 2008-2020 Frank Eskesen.
//
//       This file is free content, distributed under the GNU General
//       Public License, version 3.0.
//       (See accompanying file LICENSE.GPL-3.0 or the original
//       contained within https://www.gnu.org/licenses/gpl-3.0.en.html)
//
//----------------------------------------------------------------------------
//
// Title-
//       SigmaHoleInfo.java
//
// Purpose-
//       Sigma hole information container class.
//
// Last change date-
//       2020/01/15
//
//----------------------------------------------------------------------------
import java.awt.event.FocusListener;
import java.lang.*;
import java.util.*;
import javax.swing.*;

//----------------------------------------------------------------------------
//
// Class-
//       SigmaHoleInfo
//
// Purpose-
//       Sigma hole information container class.
//
//----------------------------------------------------------------------------
public class SigmaHoleInfo  extends HoleInfo {
//----------------------------------------------------------------------------
//
// Method-
//       SigmaHoleInfo.SigmaHoleInfo
//
// Purpose-
//       Constructors.
//
//----------------------------------------------------------------------------
   SigmaHoleInfo(                   // Constructor
     int               format,      // The HolePanel format
     String            title,       // The ID string, iff editable
     String[]          array)       // The hole data array
{
   super(format, title, array);
}

   SigmaHoleInfo(                   // Constructor
     int               format,      // The HolePanel format
     String            title,       // The ID string, iff editable
     String            string)      // The hole String
{
   super(format, title, string);
}

   SigmaHoleInfo(                   // Copy constructor
     SigmaHoleInfo     source)      // Source SigmaHoleInfo
{
   super(source);
}

//----------------------------------------------------------------------------
//
// Method-
//       SigmaHoleInfo.genPanel
//
// Purpose-
//       Generate the internal PANEL, optional FocusListener and NextFocusAdapter
//
//----------------------------------------------------------------------------
public synchronized DataField       // Resultant tail DataField
   genPanel(                        // Generate new Panel
     FocusListener     listener,    // (OPTIONAL) FocusListener
     DataField         tail)        // (OPTIONAL) The current tail DataField
{
   // The panel is always generated
   panel= new SigmaHolePanel(format, title, array);
   DataField[] df= panel.getField();
   if( listener != null )
   {
     if( title != null )
     {
       df[HOLE_ID].addFocusListener(listener);
       df[HOLE_ID].setEditable(true);
     }

     for(int hole= 1; hole<=18; hole++)
     {
       df[hole].addFocusListener(listener);
       df[hole].setEditable(true);
     }
   }

   if( tail != null )
   {
     tail.addFocusListener(new NextFocusAdapter(title != null ? df[HOLE_ID] : df[1]));
     df[9].addFocusListener(new NextFocusAdapter(df[10]));
     tail= df[18];
   }

   return tail;
}
} // class SigmaHoleInfo

File: src/py/Golf/SigmaHoleInfo.py
##############################################################################
##
##       Copyright (C) 2019 Frank Eskesen.
##
##       This file is free content, distributed under the GNU General
##       Public License, version 3.0.
##       (See accompanying file LICENSE.GPL-3.0 or the original
##       contained within https://www.gnu.org/licenses/gpl-3.0.en.html)
##
##############################################################################
##
## Title-
##       SigmaHoleInfo.py
##
## Purpose-
##       Golf: Hole information, with OUT/IN/TOTAL
##
## Last change date-
##       2019/08/21
##
##############################################################################
import sys

#### PyQt5 ###################################################################
#### PyQt5.QtCore    import *
#### PyQt5.QtGui     import *
#### PyQt5.QtWidgets import *

#### lib #####################################################################
#### lib.Debug       import *
#### lib.Utility     import *

#### Golf ####################################################################
from GolfApplet      import *
from HoleInfo        import *

##############################################################################
##
## Class-
##       SigmaHoleInfo
##
## Purpose-
##       HoleInfo, computing HOLE_OUT, HOLE_IN, and HOLE_TOT
##
## Usage-
##       (Same constructor as HoleInfo)
##
##############################################################################
class SigmaHoleInfo(HoleInfo):
    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)

    def __repr__(self):
        return '<SigmaHoleInfo(%s)>' % (self.holes[:FORMAT_MIN])

    def setText(self, index, value):
        super().setText(index, value)
        self.sigma()

    def sigma(self):
        print('sigma')
        out = 0
        try:
            for hole in range(1, 10):
                out += int(self.get(hole))
        except:
            out = 'N/A'
        super().setText(HOLE_OUT, str(out))

        inp = 0
        try:
            for hole in range(10, 19):
                inp += int(self.getText(hole))
        except:
            inp = 'N/A'
        super().setText(HOLE_IN, str(inp))

        tot = 0
        try:
            tot = str(int(out) + int(inp))
        except:
            tot = 'N/A'
        super().setText(HOLE_TOT, str(tot))

